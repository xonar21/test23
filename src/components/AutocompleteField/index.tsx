import React, { useState, useEffect } from "react";
import { FormControl, TextField, CircularProgress, Autocomplete, styled } from "@mui/material";
import { useDispatch } from "react-redux";
import { SelectListAction, SelectListSelector, SelectListsActions } from "~/store";

interface AutocompleteFieldProps {
  field: any;
  fieldValue: any;
  fieldKey: string;
  setValue: (newValue: any) => void;
  value: any;
  dinamicTitle: string;
  t: any;
  openModal: boolean;
  selectList: any;
  paramsRequest: any;
}

const AutocompleteField: React.FC<AutocompleteFieldProps> = ({
  field,
  fieldValue,
  fieldKey,
  setValue,
  value,
  dinamicTitle,
  t,
  openModal,
  selectList,
  paramsRequest,
}) => {
  const dispatch = useDispatch();
  const [page, setPage] = useState(1);
  const [loading, setLoading] = useState(false);
  const [totalPages, setTotalPages] = useState(0);
  const [searchValue, setSearchValue] = useState("");
  const actionKey: SelectListAction = field.action;
  const action = SelectListsActions[actionKey];
  const selectkey: SelectListSelector = field.keySelect;
  const [items, setItems] = useState(selectList[selectkey].items);
  const StyledListbox = styled("ul")({
    "&::-webkit-scrollbar": {
      width: "4px",
    },
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "rgba(0, 48, 92, 1)",
      borderRadius: "4px",
    },
    "&::-webkit-scrollbar-track": {
      backgroundColor: "rgba(230, 230, 230, 1)",
    },
  });

  useEffect(() => {
    const fetchData = async () => {
      if (field.action && !field.parent) {
        await dispatch(
          action(
            paramsRequest(page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [page, searchValue]);

  useEffect(() => {
    setTotalPages(selectList[selectkey].totalPages);

    if (selectList[selectkey].totalPages === 1) {
      setItems(selectList[selectkey].items);

      return;
    }

    setItems((prevItems: any) => [...prevItems, ...selectList[selectkey].items]);
  }, [selectList[selectkey].items]);

  useEffect(() => {
    const fetchData = async () => {
      if (field.parent && value[field.parent]) {
        const paramsRequest = (
          id: string,
          number?: number,
          size?: number,
          filters?: object,
          sortField?: string,
          sortOrder?: string,
        ) => {
          return {
            id,
            number,
            size,
            filters,
            sortField,
            sortOrder,
          };
        };

        await dispatch(
          action(
            paramsRequest(value[field.parent], page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [value[field.parent]]);

  useEffect(() => {
    if (field.parent) {
      setItems([]);
    }
  }, [openModal]);

  return (
    <FormControl fullWidth>
      <Autocomplete
        id="demo-autocomplete"
        options={items}
        value={fieldValue}
        getOptionLabel={option => {
          if (dinamicTitle === t("edit")) {
            let itemValue = "";
            items.forEach((e: any) => {
              if (e.key === option) {
                itemValue = e.value;
              }
            });
            return itemValue;
          }
          return option ? option.value : "";
        }}
        onChange={(e, newValue) => {
          setValue({ ...value, [fieldKey]: newValue?.key });
        }}
        onInputChange={(event, value, reason) => {
          if (reason === "reset") {
            setPage(1);
          }
          if (reason === "input") {
            setSearchValue(value);
            setPage(1);
            setItems([]);
          }
          if (reason === "clear") {
            setPage(1);
            setSearchValue("");
          }
          if (value === "") {
            setPage(1);
            setSearchValue("");
          }
        }}
        ListboxComponent={StyledListbox}
        ListboxProps={{
          onScroll: event => {
            const target = event.target as HTMLUListElement;
            const { scrollTop, scrollHeight, clientHeight } = target;
            if (scrollHeight - scrollTop === clientHeight) {
              setPage(prevPage => {
                if (prevPage !== totalPages) {
                  return prevPage + 1;
                }
                return prevPage;
              });
            }
          },
        }}
        renderInput={params => (
          <TextField
            {...params}
            label={t(`${field.fieldName}`)}
            InputLabelProps={{ ...params.InputLabelProps }}
            InputProps={{
              ...params.InputProps,
              endAdornment: (
                <>
                  {loading ? <CircularProgress size={20} /> : null}
                  {params.InputProps.endAdornment}
                </>
              ),
            }}
          />
        )}
        renderOption={(props, option) => <li {...props}>{option.value}</li>}
      />
    </FormControl>
  );
};

export default AutocompleteField;
