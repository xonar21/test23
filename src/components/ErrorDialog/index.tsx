import React from "react";
import Dialog from "@mui/material/Dialog";
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogActions from "@mui/material/DialogActions";
import Button from "@mui/material/Button";

interface ErrorDialogProps {
  open: boolean;
  onClose: () => void;
  errorText: string;
}

const ErrorDialog: React.FC<ErrorDialogProps> = ({ open, onClose, errorText }) => {
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>{"Ошибка"}</DialogTitle>
      <DialogContent>
        <DialogContentText>{errorText}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} color="primary">
          Закрыть
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ErrorDialog;
