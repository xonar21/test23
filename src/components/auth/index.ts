export * from "./auth-button";
export * from "./auth-form";
export * from "./auth-provider";
