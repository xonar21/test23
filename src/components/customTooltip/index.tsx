import React, { useMemo } from "react";
import { ColDef, ITooltipParams } from "ag-grid-community";
import { Box } from "@mui/material";

type CustomTooltipParams = ITooltipParams & {
  color: string;
  colDef: ColDef;
  shouldShowTooltip: boolean;
  rowData: any;
};
export default (props: CustomTooltipParams) => {
  const data = useMemo(() => props.api.getDisplayedRowAtIndex(props.rowIndex!)!.data, []);

  if (props && props.column) {
    if ("colId" in props.column) {
      const cellSelector = `.ag-row[row-id="${props.rowIndex}"] .ag-cell[col-id="${props.column.getColId()}"]`;
      const cellElement = document.querySelector(cellSelector);
      if (cellElement && cellElement) {
        if (cellElement.scrollWidth === cellElement.clientWidth) {
          return null;
        }
      }
    }
  }

  const subProperties = (data: any, props: any) => {
    const [firstField, secondField] = props.colDef.field.split(".");

    if (secondField) {
      return data[firstField][secondField];
    }

    return data[firstField];
  };
  return (
    <>
      {props.colDef.field ? (
        <Box
          sx={{
            backgroundColor: props.color || "white",
            border: "1px solid rgba(0, 48, 92, 1)",
            minWidth: "100px",
            borderRadius: "3px",
            padding: "5px",
            transition: "all 1s linear",
            maxWidth: "300px",
            whiteSpace: "pre-wrap !important",
            overflowWrap: "break-word !important",
            boxSizing: "border-box !important",
          }}
        >
          {subProperties(data, props)}
          {/* {data[props.colDef.field]} */}
        </Box>
      ) : (
        <></>
      )}
    </>
  );
};
