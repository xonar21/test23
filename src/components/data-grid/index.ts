export { default as DataGrid } from "./DataGrid";
export * from "./configs";
