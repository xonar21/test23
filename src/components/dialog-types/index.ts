export { default as DialogTypes } from "./dialog-types";
export * from "./dialog-types";
