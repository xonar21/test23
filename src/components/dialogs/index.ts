export * from "./prompt-dialog";
export * from "./post-form-dialog";
export * from "./dialog-provider";
