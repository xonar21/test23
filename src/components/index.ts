export * from "./auth";
export * from "./breadcrumbs";
export * from "./data-grid";
export * from "./demo";
export * from "./dialogs";
export * from "./dialog-types";
export * from "./formik";
export * from "./header";
export * from "./mui";
export * from "./navigation-drawer";
export * from "./page-loading-indicator";
export * from "./seo";
export * from "./snackbar-provider";
