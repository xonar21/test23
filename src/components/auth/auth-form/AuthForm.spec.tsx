import { AuthForm } from "~/components";
import { AuthActionType } from "~/models";
import { render, testIds, fireEvent, act } from "~/shared";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("next/router", () => ({
  ...jest.requireActual("next/router"),
  useRouter: () => ({
    query: {},
  }),
}));

describe("Auth Form component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<AuthForm type="login" />);
    expect(asFragment()).toMatchSnapshot();
  });

  const actions: AuthActionType[] = ["login", "register"];
  const payloads = [
    {
      succeeded: true,
      payload: { token: "test" },
    },
    {
      succeeded: false,
      payload: { data: { error: "Test error" } },
    },
  ];

  actions.forEach(action => {
    payloads.forEach(payload => {
      it(`should ${action} with succeeded: ${payload.succeeded}`, async () => {
        mockDispatch.mockReturnValue(payload);

        const { getByTestId } = render(<AuthForm type={action} />);
        const submitButton = getByTestId(testIds.components.authForm.submitButton);
        const emailInput = getByTestId(testIds.components.authForm.email).querySelector("input") as HTMLInputElement;
        const passwordInput = getByTestId(testIds.components.authForm.password).querySelector(
          "input",
        ) as HTMLInputElement;

        await act(async () => {
          fireEvent.input(emailInput, { target: { value: "johndoe@domain.com" } });
          fireEvent.input(passwordInput, { target: { value: "Pa$$w0rd" } });
        });

        await act(async () => {
          fireEvent.click(submitButton);
        });
      });
    });
  });
});
