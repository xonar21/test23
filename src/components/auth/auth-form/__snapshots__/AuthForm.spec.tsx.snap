// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Form component should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
    novalidate=""
  >
    <div
      class="MuiCardContent-root css-1xc3x24-MuiCardContent-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ht8sh5-MuiTypography-root"
      >
        auth.login
      </h5>
      <div
        class="MuiBox-root css-i3pbo"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="auth-form-email-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-1"
            id="mui-1-label"
          >
            auth.email
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-1"
              name="email"
              required=""
              type="text"
              value="george.bluth@reqres.in"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  auth.email *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          data-testid="auth-form-password-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-2"
            id="mui-2-label"
          >
            auth.password
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-2"
              name="password"
              required=""
              type="password"
              value="Pa$$w0rd"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  auth.password *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-w4brw1-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-1fdrpn6-MuiButtonBase-root-MuiButton-root"
        data-testid="auth-form-submit-button"
        tabindex="0"
        type="submit"
      >
        <div
          class="MuiBox-root css-ysinle"
        >
          auth.login
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover css-1scu8px-MuiTypography-root-MuiLink-root"
        href="/auth/register"
      >
        auth.redirectFrom.login
      </a>
    </div>
  </form>
</DocumentFragment>
`;
